full-qr-powered-lanes-demo/query-rules-lanes/query-rules-lanes.js

```diff
@@ -11,6 +8,7 @@ const queryRulesLanes = instantsearch.connectors.connectQueryRules(
       return;
     }
 
+    const state = getState(widgetParams.container);
```

Comment:
I guess it's fine since it's not a production-ready widget but this can cause collisions if people rely on CSS classes as selectors. Another idea would be to have a global counter as the state key.

Reply:
@francoischalifour How can there be collisions if they use CSS classes as selectors?
Can you give me an example? I thought if they put `container: "#abc"`, then `"#abc"` would work fine as a key for `state`.

If we can use a global counter, we could do it inside the `render` function passed to the connect by using `isFirstRender` to increment the counter only once initially. Is this what you had it mind?

Reply:
In this widget it's fine because we only support strings but imagine you pass a DOM element to the `container` option (as we do for officially supported widgets), suddenly the key is `[object HTMLDivElement]`. This key won't be unique anymore.

The global counter would be incremented in `getState` when `states[key]` doesn't exist.

Either way, it's fine for this widget until we support DOM containers.

Reply:
Ahh okay. It can cause problems if users pass DOM element to the container option. I get it.
I will leave it as is for now.